Files/Models/JsonSettings/Implementation/CachingJsonSettingsDatabase.cs

```diff
@@ -13,6 +13,26 @@ public CachingJsonSettingsDatabase(IJsonSettingsSerializer jsonSettingsSerialize
         {
         }
 
+        public override TValue GetValue<TValue>(string key, object defaultValue = null)
+        {
+            if (settingsCache.ContainsKey(key))
+            {
+                var value = settingsCache[key];
+                if (value is Newtonsoft.Json.Linq.JToken jTokenValue)
+                {
+                    var objValue = jTokenValue.ToObject<TValue>();
+                    settingsCache[key] = objValue;
```

Comment:
For file tags I need `GetValue` to return the same instance of an object on subsequent calls, so I cache the actual value instead of the JToken.